defrag/modules/obs_packages.py

```diff
@@ -0,0 +1,112 @@
+from operator import attrgetter
+from typing import Any, Dict, Generator, List
+from aiohttp.helpers import BasicAuth
+from defrag.modules.helpers.requests import Session
+from requests import Request
+from dataclasses import dataclass
+import lxml.etree
+
+from defrag import BUGZILLA_PASSWORD, BUGZILLA_USER
+
+__MOD_NAME__ = "obs_packages"
+
+
+@dataclass
+class PreparedRequest:
+    url: str
+
+
+@dataclass
+class OS_API_QUERY:
+    base_url: str
+    api_path: str
+    distribution: str
+    match_keyword: str
+
+    def build(self) -> PreparedRequest:
+        url_endpoint = self.base_url + self.api_path
+        xpath = "contains-ic(@name, '%s') and path/project='%s'" % (
+            self.match_keyword, self.distribution)
+        prep_req = Request('GET', url_endpoint, params={
+                           'match': xpath, "limit": 0}).prepare().url
+        if not prep_req:
+            raise Exception("Unable to build prepared request!")
+        return PreparedRequest(prep_req)
+
+
+@dataclass
+class PackageEntry:
+    arch: str
+    baseproject: str
+    filepath: str
+    name: str
+    package: str
+    project: str
+    release: str
+    repository: str
+    version: str
+
+
+def build(items: List[Dict[str, Any]]) -> Generator[PackageEntry, None, None]:
+    names = []
+    keys = PackageEntry.__annotations__.keys()
+    for i in items:
+        name = i['name']
+        if not name in names:
+            names.append(name)
+            yield PackageEntry(**{k: v for k, v in i.items() if k in keys})
+
+
+def is_relevant(entry: PackageEntry, home_repos: bool) -> bool:
```

Comment:
Packages from devel projects (such as multimedia:apps) are required since this should be used in opi in the future.

Reply:
Fixed.